crates/core/src/dht/actions/pop_next_holding_workflow.rs

```diff
@@ -4,10 +4,43 @@ use crate::{
     instance::dispatch_action,
     scheduled_jobs::pending_validations::PendingValidation,
 };
+use futures::{future::Future, task::Poll};
+use std::{pin::Pin, sync::Arc};
 
-use std::sync::Arc;
-
-pub fn pop_next_holding_workflow(pending: PendingValidation, context: Arc<Context>) {
-    let action_wrapper = ActionWrapper::new(Action::PopNextHoldingWorkflow(pending));
+pub async fn pop_next_holding_workflow(pending: PendingValidation, context: Arc<Context>) {
+    let action_wrapper = ActionWrapper::new(Action::PopNextHoldingWorkflow(pending.clone()));
     dispatch_action(context.action_channel(), action_wrapper.clone());
+    PopNextHoldingWorkflowFuture { context, pending }.await
+}
+
+pub struct PopNextHoldingWorkflowFuture {
+    context: Arc<Context>,
+    pending: PendingValidation,
+}
+
+impl Future for PopNextHoldingWorkflowFuture {
+    type Output = ();
+
+    fn poll(self: Pin<&mut Self>, cx: &mut std::task::Context) -> Poll<Self::Output> {
+        //
+        // TODO: connect the waker to state updates for performance reasons
+        // See: https://github.com/holochain/holochain-rust/issues/314
+        //
+        cx.waker().clone().wake();
+
+        if let Some(state) = self.context.try_state() {
+            match state.dht().next_queued_holding_workflow() {
```

Comment:
Would be great to make `next_queued_holding_workflow` return an address, because this is going to clone a bunch of data on every poll, which could add up